fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -3328,23 +3328,28 @@ void DDTeamCollection::updateCpuPivots() {
 }
 
 void DDTeamCollection::updateTeamEligibility() {
-	int healthyCount = 0, lowDiskCount = 0, lowCpuCount = 0;
+	int healthyCount = 0, lowDiskUtilOnly = 0, lowCpuOnly = 0, lowDiskUtilLowCpu = 0;
 	for (auto& team : teams) {
 		if (team->isHealthy()) {
 			healthyCount++;
 			if (team->hasHealthyAvailableSpace(teamPivots.pivotAvailableSpaceRatio)) {
-				lowDiskCount++;
+				lowDiskUtilOnly++;
 				team->increaseEligibilityCount(data_distribution::EligibilityCounter::LOW_DISK_UTIL);
 			} else {
 				team->resetEligibilityCount(data_distribution::EligibilityCounter::LOW_DISK_UTIL);
 			}
 
 			if (team->hasLowerCpu(teamPivots.pivotCPU)) {
-				lowCpuCount++;
+				lowCpuOnly++;
 				team->increaseEligibilityCount(data_distribution::EligibilityCounter::LOW_CPU);
 			} else {
 				team->resetEligibilityCount(data_distribution::EligibilityCounter::LOW_CPU);
 			}
+
+			if (team->hasHealthyAvailableSpace(teamPivots.pivotAvailableSpaceRatio) &&
+			    (team->hasLowerCpu(teamPivots.pivotCPU))) {
+				lowDiskUtilLowCpu++;
```

Comment:
This change will triple count the teams -- we'll add an overlapped team in line 3336 to `lowDiskUtilOnly`, in line 3343 to `lowCpuOnly`, and here to `lowDiskUtilLowCpu`. Potential fixes I can think of could be to decrement the other two counts inside this if block, or to check the other condition before incrementing the counts in the previous blocks.